src/main/scala/mesosphere/marathon/core/storage/StorageModule.scala

```diff
@@ -22,68 +19,96 @@ import scala.concurrent.ExecutionContext
   * Provides the repositories for all persistable entities.
   */
 trait StorageModule {
-  def appRepository: AppRepository
+  def appRepository: ReadOnlyAppRepository
   def taskRepository: TaskRepository
   def deploymentRepository: DeploymentRepository
   def taskFailureRepository: TaskFailureRepository
+  def groupRepository: GroupRepository
+  def migration: Migration
 }
 
 object StorageModule {
-  def apply(conf: MarathonConf, metrics: Metrics, mat: Materializer, ctx: ExecutionContext,
-    scheduler: Scheduler, actorRefFactory: ActorRefFactory): StorageModule =
-    apply(StorageConfig(conf))(metrics, mat, ctx, scheduler, actorRefFactory)
+  def apply(conf: MarathonConf)(implicit metrics: Metrics, mat: Materializer, ctx: ExecutionContext,
+    scheduler: Scheduler, actorRefFactory: ActorRefFactory): StorageModule = {
+    val currentConfig = StorageConfig(conf)
+    val legacyConfig = conf.internalStoreBackend() match {
+      case TwitterZk.StoreName => Some(TwitterZk(cache = true, conf))
```

Comment:
Not 100% sure: wouldn't `currectConfig` be already `TwitterZk` in this case? 
Why would you override the cache configuration here to be always true?
As far as I can see the legacyConfig is only used in Migration for a special purpose. Perhaps legacyConfig is a pretty generic term, perhaps we can rename it to something that expresses the intent more clearly?